### ISSUE TITLE
Module docstrings

### ISSUE THREAD

**Issue by Sxmourai** [hermit-os/kernel#1078]
Should add module docstrings to all files to explain more what they do and have some links to external docs.

**Comment by mkroening** [comment]
That's true. Some modules are nicely documented, others not so much. Hopefully we'll get better in this regard over time.

Do you have any concrete issues or suggestions? :)

**Comment by Sxmourai** [comment]
I want to help a bit this project, so I will look more into the architecture of this project and I will tell you a bit more what are the ones that need the most documentation
But I've already seen MMIO and fd i.e.
